File: JavaScript Front End/src/Messages/CommandMessages/GraphBuilding/SetNumberOfCyclesOfTaskCommandMsg.js
const GraphBuildingCommandMsg = require('./GraphBuildingCommandMsg.js').GraphBuildingCommandMsg;

class SetNumberOfCyclesOfTaskCommandMsg extends GraphBuildingCommandMsg
{
    constructor(nodeId, numberOfCycles)
    {
        super();
        this.jsonMessage["@type"] = this.jsonMessage["@type"].concat("SetNumberOfCyclesOfTaskCommandMsg");
        this.jsonMessage["nodeId"] = nodeId;
        this.jsonMessage["numberOfCycles"] = numberOfCycles;
    }
}

module.exports = {SetNumberOfCyclesOfTaskCommandMsg};

File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandMessages/GraphBuilding/SetNumberOfCyclesOfTaskCommandMsg.java
package InfoSecCooker.Communication.Messages.CommandMessages.GraphBuilding;

public class SetNumberOfCyclesOfTaskCommandMsg extends GraphBuildingCommandMsg
{
    Long nodeId;
    Integer numberOfCycles;

    public SetNumberOfCyclesOfTaskCommandMsg(Long nodeId, Integer numberOfCycles)
    {
        this.nodeId = nodeId;
        this.numberOfCycles = numberOfCycles;
    }

    public Long getNodeId()
    {
        return nodeId;
    }

    public Integer getNumberOfCycles()
    {
        return numberOfCycles;
    }
}
